odoo/addons/base/models/assetsbundle.py

```diff
@@ -340,31 +340,34 @@ def dialog_message(self, message):
                 if (window.__assetsBundleErrorSeen) return;
                 window.__assetsBundleErrorSeen = true;
 
-                document.addEventListener("DOMContentLoaded", function () {
+                var loaded = function () {
+                    clearTimeout(loadedTimeout);
                     var alertTimeout = setTimeout(alert.bind(window, message), 0);
-                    if (typeof odoo === "undefined") return;
+                    var odoo = window.top.odoo;
+                    if (!odoo || !odoo.define) return;
 
                     odoo.define("AssetsBundle.ErrorMessage", function (require) {
                         "use strict";
 
-                        var base = require("web_editor.base");
+                        require('web.dom_ready');
                         var core = require("web.core");
                         var Dialog = require("web.Dialog");
 
                         var _t = core._t;
 
                         clearTimeout(alertTimeout);
 
-                        base.ready().then(function () {
-                            new Dialog(null, {
-                                title: _t("Style error"),
-                                $content: $("<div/>")
-                                    .append($("<p/>", {text: _t("The style compilation failed, see the error below. Your recent actions may be the cause, please try reverting the changes you made.")}))
-                                    .append($("<pre/>", {html: message})),
-                            }).open();
-                        });
+                        new Dialog(null, {
+                            title: _t("Style error"),
+                            $content: $("<div/>")
+                                .append($("<p/>", {text: _t("The style compilation failed, see the error below. Your recent actions may be the cause, please try reverting the changes you made.")}))
+                                .append($("<pre/>", {html: message})),
+                        }).open();
                     });
-                });
+                }
+
+                var loadedTimeout = setTimeout(loaded, 5000);
```

Comment:
why is there a hardcoded loaded timeout of this number ?
does it represent something ?

Reply:
Absolutely nothing, you just have to read the page / assets, we wait a long time. Normally it is never used "clearTimeout (loadedTimeout);" but if the page does not trigger anything or if the asset is read by ajax and there is an error it must appear at a time (and do not make an alert if you can do a dialog).